### PR TITLE
OAuth exceptions added

### PR DESCRIPTION
**Pull request by elilmatha**

### CODE REVIEW

File: components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java [hunk 1]
```diff
@@ -59,14 +60,21 @@ public Mediator createSpecificMediator(OMElement element, Properties properties)
         remoteServiceUrl = element.getAttribute(ATTR_NAME_SERVICE_EPR);
         if (remoteServiceUrl != null && remoteServiceUrl.getAttributeValue() != null) {
             mediator.setRemoteServiceUrl(remoteServiceUrl.getAttributeValue());
+        }else {
```
Comment (RKathees) [inline components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java]: Formatting issue.

File: components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java [hunk 2]
```diff
@@ -59,14 +60,21 @@ public Mediator createSpecificMediator(OMElement element, Properties properties)
...
         username = element.getAttribute(ATTR_NAME_USERNAME);
         if(username != null && username.getAttributeValue() != null){
         	mediator.setUsername(username.getAttributeValue());
+        } else {
+            throw new MediatorException("The 'username' attribute is required for the OAuth mediator ");
         }
         password = element.getAttribute(ATTR_NAME_PASSWORD);
         if(password != null && password.getAttributeValue() != null){
         	mediator.setPassword(password.getAttributeValue());
+        }else {
```
Comment (RKathees) [inline components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java]: Formatting issue.

Comment (elilmatha) [inline components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java]: Fixed in https://github.com/wso2/carbon-mediation/pull/742

File: components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java (hunk 1 above)
Comment (elilmatha) [inline components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java]: Fixed in https://github.com/wso2/carbon-mediation/pull/742

File: components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java [hunk 3]
```diff
@@ -59,14 +60,21 @@ public Mediator createSpecificMediator(OMElement element, Properties properties)
         remoteServiceUrl = element.getAttribute(ATTR_NAME_SERVICE_EPR);
         if (remoteServiceUrl != null && remoteServiceUrl.getAttributeValue() != null) {
             mediator.setRemoteServiceUrl(remoteServiceUrl.getAttributeValue());
+        }else {
+            throw new MediatorException(
+                    "The 'remoteServiceUrl' attribute is required for the OAuth mediator");
         }
         username = element.getAttribute(ATTR_NAME_USERNAME);
         if(username != null && username.getAttributeValue() != null){
```
Comment (abeykoon) [inline components/mediators/oauth/org.wso2.carbon.identity.oauth.mediator/src/main/java/org/wso2/carbon/identity/oauth/mediator/config/xml/OAuthMediatorFactory.java]: Let us use proper parentheses to separate the logic.
what abt 
 (null != username) && (null != username.getAttributeValue())

**Pull request merged**